### PR TITLE
Synchronising development and staging branches

### PR DESCRIPTION
**Pull request by augustinas**
Maintenance PR to synchronise changes between the dev and staging branches.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -91,10 +100,11 @@ script:
   - sbt ++$TRAVIS_SCALA_VERSION compile
   - |
     if [[ "$REPOSITORY_NAME" = "hubat" ]]; then
+      # TODO: Update test next
       sbt ++$TRAVIS_SCALA_VERSION test:compile
       sbt ++$TRAVIS_SCALA_VERSION "project hat" coverage test -Dconfig.file=hat/conf/application.test.conf
     fi
-  - sbt -Denv=prod docker:stage
+  - sbt -Denv=prod -Dhttp.port=8080 -Dpidfile.path=/dev/null -Dplay.server.pidfile.path=/dev/null docker:stage
```
Comment (benshaw) [inline .travis.yml:107]: what is the intent here ?
These java variables get passed to the docker:stage task, but my understanding is that they WILL NOT be set when you run the container. An option which will make them consistently available is adding them to the build.sbt i.e.  javaOptions in Universal ++= Seq("-Dhttp.port=8080")

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+
+test:
+	$(info > Building HAT2.0 for test.)
+	@sbt ${TEST_SBT_JAVA_OPT} compile
+
+
+docker-prod:
+	$(info > Building a docker image to use in production.)
+	@sbt ${PROD_DOCKER_SBT_JAVA_OPT} docker:stage
```
Comment (benshaw) [inline Makefile:26]: See above comment re the scope of java options passed in here

File: build.sbt [hunk 3]
```diff
@@ -53,9 +53,12 @@ lazy val hat = project
       case BuildEnv.Stage | BuildEnv.Production =>
         libraryDependencies.value.map(excludeSpecs2)
     }),
+    libraryDependencies += "org.codehaus.janino" % "janino" % "3.1.2",
```
Comment (benshaw) [inline build.sbt:56]: intrigued as to why this was added

File: .travis.yml (hunk 1 above)
Comment (augustinas) [reply .travis.yml:107]: This is the legacy of different attempts to debug build issues. Suggest leaving it there for the current merge as we have confirmed working build process. Once merged we should clean up on dev and propagate to other branches.

File: build.sbt (hunk 3 above)
Comment (augustinas) [reply build.sbt:56]: me too :)

Comment (benshaw) [reply build.sbt:56]: @dataswifty ?? Probably not a big deal https://github.com/Hub-of-all-Things/HAT2.0/commit/67088343b05f0051c58d142149c9fb8f4f77239c#diff-fdc3abdfd754eeb24090dbd90aeec2ceR56

**Pull request merged**